python/ray/data/dataset.py

```diff
@@ -3608,15 +3608,12 @@ def to_arrow_refs(self) -> List[ObjectRef["pyarrow.Table"]]:
         Returns:
             A list of remote Arrow tables created from this dataset.
         """
-        blocks: List[ObjectRef[Block]] = self.get_internal_block_refs()
-
-        if self.dataset_format() == BlockFormat.ARROW:
-            # Zero-copy path.
-            return blocks
-
-        block_to_arrow = cached_remote_fn(_block_to_arrow)
-        return [block_to_arrow.remote(block) for block in blocks]
+        # Convert into Arrow blocks.
+        arrow_ds = self.map_batches(lambda x: x, batch_format="pyarrow")
```

Comment:
This could be a regression. How about we launch a task to check if the first block format is already pyarrow? That should be fast since get_internal_block_refs should be materializing all blocks anyways.

Reply:
Hmm I would expect this to actually be faster since we it can leverage our stage fusion optimizations and @clarkzinzow's work to elide batch transforms.

Keeps the abstractions clean since all the execution/optimization is contained within Datasets core.

Reply:
Oh I see, I forgot we are lazy by default now :) In principle this could cause overheads for cached datasets still though, so maybe at least add a TODO here?

Reply:
reverted this back to the old behavior